src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/OnFinishedAction.java

```diff
@@ -63,7 +66,42 @@ private void sendPeriodicPingNotification(HaFlowUpdateFsm stateMachine) {
     }
 
     private void updateFlowMonitoring(HaFlowUpdateFsm stateMachine) {
-        // TODO activate/deactivate flow monitoring for multi switch flows if required
-        // https://github.com/telstra/open-kilda/issues/5172
+        HaFlow original = stateMachine.getOriginalHaFlow();
+        HaFlow target = HaFlowMapper.INSTANCE.toHaFlow(stateMachine.getTargetHaFlow());
+
+        for (HaSubFlow originalSubFlow : original.getHaSubFlows()) {
+            HaSubFlow targetSubFlow = target.getHaSubFlowOrThrowException(originalSubFlow.getHaSubFlowId());
+            boolean originalNotSingle = !originalSubFlow.isOneSwitch();
+            boolean targetNotSingle = !targetSubFlow.isOneSwitch();
+            boolean srcUpdated = isSrcUpdated(original, target);
+            boolean dstUpdated = isDstUpdated(originalSubFlow, targetSubFlow);
+
+            // clean up old if it is not single
```

Comment:
as an option, these comments could be private methods and you can filter out one-switch flows before calling these methods.

Reply:
this is a pattern saying that when you want to leave a single line comment explaining something about the small code block inside a larger code block it is much better to extract this block to a separate method and move the comment text to a method name:
```
instead: 
// clean up blah-blah
some code here

do:
private void cleanUpBlahBlah() {
some code from above here
}
```